### PR TITLE
feat(headercell): table Header Cell Implementation

### PR DESCRIPTION
**Pull request by masoudmanson**
## Summary

**HeaderCell**
Shortcut ticket: [sh-208411](https://app.shortcut.com/sci-design-system/story/208411/header-cell-component)

Implemented with &lt;th&gt; under the hood.

## Checklist

- [x] Default Story in Storybook
- [ ] LivePreview Story in Storybook
- [x] Test Story in Storybook
- [x] Tests written
- [x] Variables from `defaultTheme.ts` used wherever possible
- [ ] If updating an existing component, depreciate flag has been used where necessary
- [ ] Chromatic build verified by @chanzuckerberg/sds-design

### CODE REVIEW

**Review by clarsen-czi — APPROVED** [review]
LGTM, thanks @masoudmanson !!

File: src/core/HeaderCell/index.test.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { generateSnapshots } from "@chanzuckerberg/story-utils";
+import { composeStory } from "@storybook/testing-react";
+import { fireEvent, render, screen } from "@testing-library/react";
+import React from "react";
+import * as snapshotTestStoryFile from "./index.stories";
+import Meta, { Test as TestStory } from "./index.stories";
```
Comment (mmmmmmaya) [inline src/core/HeaderCell/index.test.tsx:6]: it seems weird there are two import lines from the same file.

**Review by mmmmmmaya — APPROVED** [review]

File: src/core/HeaderCell/index.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+import {
+  HeaderCellExtraProps,
+  StyledSortingIcon,
+  StyledTableHeader,
+} from "./style";
+
+interface HeaderCellContentProps {
+  active?: boolean;
+  textPosition?: "left" | "right";
```
Comment (bethbertozzi) [inline src/core/HeaderCell/index.tsx:13]: should we change this to include a 'center' option for consistency, if we are doing it for other cells?

Comment (masoudmanson) [reply src/core/HeaderCell/index.tsx]: Thanks for spotting this one, I added it in the final push 👍🏻

**Pull request merged**